inject.js

```diff
@@ -351,6 +351,16 @@ module.exports = function(acorn) {
     return this.finishNode(node, 'JSXElement');
   };
 
+  // Parse JSX text
+
+  pp.jsx_parseText = function(value) {
+    var node = this.startNode();
```

Comment:
Perhaps worth just calling existing `parseLiteral` and changing `.type` on result?